pkg/asset/ignition/bootstrap/cvoignore.go

```diff
@@ -0,0 +1,123 @@
+}
+
+// Name returns a human friendly name for the operator
+func (a *CVOIgnore) Name() string {
+	return "CVO Ignore"
+}
+
+// Dependencies returns all of the dependencies directly needed by the CVOIgnore asset
+func (a *CVOIgnore) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&manifests.Manifests{},
+		&manifests.Openshift{},
+	}
+}
+
+// Generate generates the respective operator config.yml files
+func (a *CVOIgnore) Generate(dependencies asset.Parents) error {
+	operators := &manifests.Manifests{}
+	openshiftManifests := &manifests.Openshift{}
+	dependencies.Get(operators, openshiftManifests)
+
+	var clusterVersion *unstructured.Unstructured
+	var ignoredResources []interface{}
+	var files []*asset.File
+	files = append(files, operators.FileList...)
+	files = append(files, openshiftManifests.FileList...)
+	for _, file := range files {
+		u := &unstructured.Unstructured{}
+		if err := yaml.Unmarshal(file.Data, u); err != nil {
+			return errors.Wrapf(err, "could not unmarshal %q", file.Filename)
+		}
+		if file.Filename == cvoOverridesFilename {
+			clusterVersion = u
+			continue
+		}
+		ignoredResources = append(ignoredResources,
+			configv1.ComponentOverride{
+				Kind:      u.GetKind(),
+				Group:     u.GetObjectKind().GroupVersionKind().Group,
+				Namespace: u.GetNamespace(),
+				Name:      u.GetName(),
+				Unmanaged: true,
+			})
+	}
+
+	specAsInterface, ok := clusterVersion.Object["spec"]
+	if !ok {
+		specAsInterface = map[string]interface{}{}
+		clusterVersion.Object["spec"] = specAsInterface
+	}
+	spec, ok := specAsInterface.(map[string]interface{})
+	if !ok {
+		return errors.Errorf("unexpected type (%T) for .spec in clusterversion", specAsInterface)
+	}
+	originalOverridesAsInterface := spec["overrides"]
+	originalOverrides, ok := originalOverridesAsInterface.([]interface{})
+	if !ok && originalOverridesAsInterface != nil {
+		return errors.Errorf("unexpected type (%T) for .spec.overrides in clusterversion", originalOverridesAsInterface)
+	}
+	originalOverridesPatch := map[string]interface{}{
+		"spec": map[string]interface{}{
+			"overrides": originalOverrides,
+		},
+	}
+	spec["overrides"] = append(ignoredResources, originalOverrides...)
+
+	cvData, err := yaml.Marshal(clusterVersion)
+	if err != nil {
+		return errors.Wrap(err, "error marshalling clusterversion")
+	}
+	a.FileList = append(a.FileList, &asset.File{
+		Filename: cvoOverridesFilename,
+		Data:     cvData,
+	})
+
+	origOverrideData, err := json.Marshal(originalOverridesPatch)
+	if err != nil {
+		return errors.Wrap(err, "error marshalling original overrides")
+	}
+	a.FileList = append(a.FileList, &asset.File{
+		Filename: originalOverridesFilename,
+		Data:     origOverrideData,
+	})
+
+	return nil
+}
+
+// Files returns the files generated by the asset.
+func (a *CVOIgnore) Files() []*asset.File {
+	return a.FileList
+}
+
+// Load does nothing as the file should not be loaded from disk.
+func (a *CVOIgnore) Load(f asset.FileFetcher) (bool, error) {
```

Comment:
it seems like it might be possible to avoid the `this must come after manifests.Manifests so that it replaces cvo-overrides.yaml` tweak by having this asset directly read that particular manifest from disk.  But perhaps the order-based clobber is more straightforward after all?

Reply:
If the asset reads the manifest from disk, then it will write the manifest to disk, too. Unfortunately, that is problematic because it gives the user an opportunity to make changes to the manifest. Also, I am not sure that it wouldn't cause the original manifest asset to become dirty and create a never-ending cycle.